### ISSUE TITLE
Σημαντικό overfit στο CNN για classification

### ISSUE THREAD

**Issue by dion11ar** [slp-ntua/patrec-labs#153]
Καλημέρα, 
Στην εκπαίδευση του νευρωνικού δικτύου έχουμε το εξής πρόβλημα. Οταν το εκπαιδεύουμε, το loss στο train set γίνεται σημαντικά μικρότερο από αυτό στο validation set (στην 10η εποχή για παράδειγμα οι αντίστοιχες τιμές είναι 0.53 και 2.03). Δοκιμάσαμε να χρησιμοποιήσουμε l2 regularization καθώς και  dropout αλλά το πρόβλημα εξακολουθεί να υφίσταται (αν και σε μικρότερο βαθμό) οπότε υποθέτουμε ότι φταίει κάτι στην υλοποίηση μας αλλά δεν μπορούμε να βρούμε τι είναι αυτό.
Γνωρίζετε τι θα μπορούσε να προκαλεί αυτή την συμπεριφορά;

**Comment by artemis420** [comment]
το ιδιο θεμα εχω κ εγω .το validation δεν ΄πεφτει κατω απο 2

**Comment by efthymisgeo** [comment]
1) Βαλτε καποιο regularization π.χ dropout η weight decay και παιξτε λιγο με τις υπερπαραμετρους. Προσοχη το dropout δεν το πολυβαζουμε γενικα σε CNNs. Αλλη ιδεα ειναι καποιο residual connection, δειτε το resnet paper π.χ (αυτο βεβαια αλλαζει την αρχιτεκτονική σας)

2) Βαλτε πιο μικρο/μεγαλο CNN απο αποψη παραμετρων

**Comment by dion11ar** [comment]
Εμας λύθηκε τελικά προσθέτοντας άλλα δύο fully connected layers (με ReLU ανάμεσα) αντί για ένα μόνο που είχαμε αρχικά
